### ISSUE TITLE
kmidiff called with wrong args

### ISSUE THREAD

**Issue by dcantrell** [rpminspect/rpminspect#871]
latest rhel-9.2 165.el9 kernel testing reports kmidiff failure

https://dashboard.osci.redhat.com/#/artifact/brew-build/aid/47804263?focus=tc:osci.brew-build.rpminspect.static-analysis

Looking at the artifacts, it looks like older rpminspect version is used compared to testing of previous 164.el9

http://artifacts.osci.redhat.com/testing-farm/1cb211f1-918b-4132-ad03-2b194a7d14a0/ 

```
rpminspect profile: none
new build: kernel-5.14.0-165.el9
old build: kernel-5.14.0-162.3.1.el9 (found in rhel-9.1.0-et-compose brew tag)
Test description:
kmidiff compares the binary Kernel Module Interfaces of two Linux kernel
trees.  The binary KMI is the interface that the Linux kernel exposes to
its modules.  The trees we are interested in here are the result of the
build of the Linux kernel source tree.  If the builds compared are not
considered a rebase, an incompatible change reported by kmidiff is
reported for verification.
======================================== Test Output ========================================
kmidiff:
--------
Result: VERIFY
1) KMI comparison ended unexpectedly.
Waiver Authorization: Anyone
Details:
Command: kmidiff /var/tmp/rpminspect/kernel-5.14.0.ywixn5/before/aarch64/bpftool-debuginfo-5.14.0-162.3.1.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/before/aarch64/perf-debuginfo-5.14.0-162.3.1.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/before/aarch64/kernel-debuginfo-5.14.0-162.3.1.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/before/aarch64/kernel-debug-debuginfo-5.14.0-162.3.1.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/before/aarch64/python3-perf-debuginfo-5.14.0-162.3.1.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/before/aarch64/kernel-tools-debuginfo-5.14.0-162.3.1.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/after/aarch64/bpftool-debuginfo-5.14.0-165.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/after/aarch64/perf-debuginfo-5.14.0-165.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/after/aarch64/kernel-debuginfo-5.14.0-165.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/after/aarch64/kernel-debug-debuginfo-5.14.0-165.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/after/aarch64/python3-perf-debuginfo-5.14.0-165.el9.aarch64/usr/lib/debug/ /var/tmp/rpminspect/kernel-5.14.0.ywixn5/after/aarch64/kernel-tools-debuginfo-5.14.0-165.el9.aarch64/usr/lib/debug/ --vmlinux1 /var/tmp/rpminspect/kernel-5.14.0.ywixn5/before/aarch64/kernel-debuginfo-5.14.0-162.3.1.el9.aarch64//usr/lib/debug/lib/modules/5.14.0-162.3.1.el9.aarch64/vmlinux --vmlinux2 /var/tmp/rpminspect/kernel-5.14.0.ywixn5/after/aarch64/kernel-debuginfo-5.14.0-165.el9.aarch64//usr/lib/debug/lib/modules/5.14.0-165.el9.aarch64/vmlinux /var/tmp/rpminspect/kernel-5.14.0.ywixn5/before/aarch64/kernel-debuginfo-5.14.0-162.3.1.el9.aarch64 /var/tmp/rpminspect/kernel-5.14.0.ywixn5/after/aarch64/kernel-debuginfo-5.14.0-165.el9.aarch64
Exit code: 1
kmidiff: unrecognized option: 
try the --help option for more information
Suggested Remedy:
Kernel Module Interface introduced during maintenance updates can lead to problems for users.  See the libabigail documentation and the distribution KMI policy to determine if this detected chan
```

kernel-5.14.0-165.el9 - rpminspect version: 1.11-0.1.202207201958gitc3b553e.el9 - fails with unrecognized option:
kernel-5.14.0-164.el9 - rpminspect version: 1.11-0.1.202209081413git8e6ef25.el9 - OK
$ git log --oneline c3b553e..8e6ef25 lib/inspect_kmidiff.c
09d58fc [fix] Use add_abidiff_arg() in the kmidiff inspection
10e0bcb [lib] Rename abidiff-specific helper functions
There was obviously change how to args are setup between these two versions for kmidiff.

**Comment by dcantrell** [comment]
This looks like the rpminspect runner for gating has an older version of libabigail.  I updated the option generation in librpminspect, but that seems to have broken support for older versions of libabigail.  For reference, I have libabigail 2.0.0 on my workstation while the job linked above was using an environment with libabigail 1.8.2.

Working on a patch.

**Comment by dcantrell** [comment]
I do not know what the runner is using.  @msrb can you help here?  It looks like an el9 runner, but epel-9 has libabigail-2.0.0.  What is the runner using now?  I had to go back to Fedora 32 to find a released version of Fedora with a 1.8.x version of libabigail.

**Comment by dcantrell** [comment]
OK, just for completeness... I no longer think this is a bug.  It was unforunate that the kernel team saw it in production while I was working on the fix.  The commits where I was reworking how rpminspect invokes abidiff and kmidiff started on 21-Jul-2022.  This bug report shows a kmidiff failure that's using a build of rpminspect from 20-Jul-2022.  I am unable to reproduce this problem on recent builds and I verified that options on kmidiff on 1.8.2 and 1.8.1 are still the same as they are on 2.x.x.
